R/print-str.R

```diff
@@ -58,6 +104,23 @@ obj_print_footer.default <- function(x, ...) {
   invisible(x)
 }
 
+local_max_print <- function(max, frame = parent.frame()) {
+  max_print <- getOption("max.print")
+  if (is.null(max)) {
+    max <- max_print
+  }
+
+  stopifnot(is_integerish(max, 1L, finite = TRUE), max >= 0, max < 2147483648)
+  max <- as.integer(max)
+
+  if (max > max_print) {
+    # Avoid truncation in case we're forwarding to print()
+    local_options(max.print = max, .frame = frame)
```

Comment:
1. Is there any downside to unconditionally setting `max.print` to simplify things?

2. Can you please try and move the `local_options()` to the generic and rename this function to `as_max_print()`. This would simplify things.